system-addon/content-src/components/TopSites/TopSiteFormInput.jsx

```diff
@@ -19,17 +35,21 @@ export class TopSiteFormInput extends React.PureComponent {
 
   render() {
     const showClearButton = this.props.value && this.props.onClear;
-    const {validationError, typeUrl} = this.props;
+    const {typeUrl} = this.props;
+    const {validationError} = this.state;
 
     return (<label><FormattedMessage id={this.props.titleId} />
       <div className={`field ${typeUrl ? "url" : ""}${validationError ? " invalid" : ""}`}>
-        {showClearButton &&
-          <div className="icon icon-clear-input" onClick={this.props.onClear} />}
+        {this.props.loading ?
+          <div className="loading-container"><div className="loading-animation" /></div> :
+          showClearButton && <div className="icon icon-clear-input" onClick={this.props.onClear} />}
         <input type="text"
           value={this.props.value}
           ref={this.onMount}
-          onChange={this.props.onChange}
-          placeholder={this.props.intl.formatMessage({id: this.props.placeholderId})} />
+          onChange={this.onChange}
+          placeholder={this.props.intl.formatMessage({id: this.props.placeholderId})}
+          autoFocus={this.props.shouldFocus}
+          disabled={this.props.loading} />
```

Comment:
Did design confirm that the input should be disabled? Otherwise the only way to interrupt seems to be to click cancel for the whole dialog. Why does this need to be disabled in the first place?

Reply:
I thought that is what you were trying to show with the [gif from the comment](https://github.com/mozilla/activity-stream/pull/3891#issuecomment-372448047) but canceling the request also works. I'll check with design.

Reply:
I was showing that the wrong preview result could be shown, yes. But now that we're checking whether the response/screenshot/failure is for the requested url, it doesn't need to be disabled.

Reply:
I mentioned the concerns about canceling, the decision was to keep the disabled state.

> i think disabling it is the right thing to do
> unlikely that someone will want to change it mid-stream
> but disabling the input would prevent any accidental typos

Reply:
Sounds like design wants to prevent the user from changing it even though the user might want to change it before the 30 seconds is up.